src/main/java/uk/nhs/digital/ps/migrator/model/hippo/XlsxReader.java

```diff
@@ -0,0 +1,109 @@
+package uk.nhs.digital.ps.migrator.model.hippo;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.AbstractMap;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+/**
+ * Shared functions for streaming and reading Excel sheets/rows.
+ */
+public class XlsxReader {
+ 
+    private Map<String, Integer> headerNameIndexPairs;
```

Comment:
By default, all Spring beans (instances created in `@Bean` methods in `@Configuration` annotated class; in our case it's `MigratorConfiguration`) are effectively singletons that are shared among all objects that request them. As a consequence, either they should not have state or they should not be singletons as this introduces a risk of one caller affecting another by inadvertently changing state of the shared instance.

We could address this in at least two ways, but, arguably, the simplest is to make sure that individual instances of `XlsxReader` are created for each object that needs one. Luckily, all that is needed to achieve that is to annotate method `MigratorConfiguration#xslReader` with `@org.springframework.context.annotation.Scope("prototype")`:

```java
@Bean
@Scope("prototype")
public XlsxReader xslReader() {
    return new XlsxReader();
}    
```

Reply:
Good spot and interesting that there is a default scope. Will add.